network/network.go

```diff
@@ -1044,6 +1059,26 @@ func (n *network) connectTo(ip utils.IPDesc) {
 	}
 }
 
+// assumes the stateLock is not held
+// returns true if the network should keep trying to connect to the IP
+// represented by [str]
+func (n *network) shouldConnect(str string) bool {
```

Comment:
Can we combine this and `shouldTrack` and just add an argument to specify whether the state lock is held?